src/traces/sankey/plot.js

```diff
@@ -124,7 +124,8 @@ module.exports = function plot(gd, calcData) {
     };
 
     var linkHoverFollow = function(element, d) {
-
+        var trace = gd._fullData[d.traceId];
+        var ptNumber = d.originalIndex;
```

Comment:
Yes your conclusion sounds good to me. Indeed it'll be an unlikely occurrence that the number of nodes and links match.